### PR TITLE
Core/Spells: Fixed Finger of Frost / Frozen Touch

### PR DESCRIPTION
**Pull request by Ryzen-0**
Triggered from "Frostbolt" and "Frozen Orb" spells.

**Changes proposed:**

-  "Finger of Frost" and "Frozen Touch" handler

**Target branch(es):** master
- [ ] 3.3.5
- [x] master

**Tests performed:**

Builded and tested in-game.

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] Better using GetEffect(EFFECT_0)->CalcValue() for % chance to proc, but it returns wrong values; probably i dont known something about it.

### CODE REVIEW

File: src/server/scripts/Spells/spell_mage.cpp [hunk 1]
```diff
@@ -695,6 +698,64 @@ class spell_mage_water_elemental_freeze : public SpellScript
...
+class spell_mage_finger_of_frost : public SpellScriptLoader
+{
+public:
+    spell_mage_finger_of_frost() : SpellScriptLoader("spell_mage_finger_of_frost") { }
+
+    class spell_mage_finger_of_frost_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_mage_finger_of_frost_SpellScript);
+
+        bool Validate(SpellInfo const* spellInfo) override
```
Comment (illfated) [inline src/server/scripts/Spells/spell_mage.cpp:713]:
```suggestion
        bool Validate(SpellInfo const* /*spellInfo*/) override
```
Need to comment out unused arguments to pass the compiler warnings (treated as error).

File: src/server/scripts/Spells/spell_mage.cpp [hunk 2]
```diff
@@ -695,6 +698,64 @@ class spell_mage_water_elemental_freeze : public SpellScript
...
+            return ValidateSpellInfo({ SPELL_MAGE_FINGERS_OF_FROST });
+        }
+
+        void HandleAfterCast()
+        {
+            Unit* caster = GetCaster();
+            uint32 spellId = GetSpell()->GetSpellInfo()->Id;
+            uint32 chance = 0;
+
+            SpellInfo const* fingerFrostInfo = sSpellMgr->GetSpellInfo(SPELL_MAGE_FINGERS_OF_FROST, GetCastDifficulty());
```
Comment (illfated) [inline src/server/scripts/Spells/spell_mage.cpp:724]:
```cpp
TrinityCore/TrinityCore/src/server/scripts/Spells/spell_mage.cpp:724:30: fatal error: 
      unused variable 'fingerFrostInfo' [-Wunused-variable]
            SpellInfo const* fingerFrostInfo = sSpellMgr->GetSpellInfo(S...
                             ^
```
Please note that the build checks stop looking for more build warnings (treated as error) once one has been found.
```suggestion
            SpellInfo const* /*fingerFrostInfo*/ = sSpellMgr->GetSpellInfo(SPELL_MAGE_FINGERS_OF_FROST, GetCastDifficulty());
```

File: src/server/scripts/Spells/spell_mage.cpp [hunk 3]
```diff
@@ -695,6 +698,64 @@ class spell_mage_water_elemental_freeze : public SpellScript
     }
 };
 
+/* 112965 - Finger of Frost
+   228597 - Frostbolt
+   84721  - Frozen Orb */
+class spell_mage_finger_of_frost : public SpellScriptLoader
+{
+public:
+    spell_mage_finger_of_frost() : SpellScriptLoader("spell_mage_finger_of_frost") { }
```
Comment (Ovahlord) [inline src/server/scripts/Spells/spell_mage.cpp:707]: drop this part and go with the RegisterSpellScript macro like all other spell scripts do

Comment (Ryzen-0) [reply src/server/scripts/Spells/spell_mage.cpp]: Done and tested. Next ones i will use same, thanks! ;)

**Comment by illfated** [comment]
Tip for GitHub comments, especially when you use a task list: use the regular character x or X inside the brackets (not the asterisk `*` ), or put a tick in the box after you have saved the comment:

- [X] list item one
- [ ] list item two

```
- [X] list item one
- [ ] list item two

```

File: src/server/scripts/Spells/spell_mage.cpp [hunk 4]
```diff
@@ -695,6 +698,53 @@ class spell_mage_water_elemental_freeze : public SpellScript
...
+    void HandleAfterCast()
+    {
+        Unit* caster = GetCaster();
+        uint32 spellId = GetSpell()->GetSpellInfo()->Id;
+        uint32 chance = 0;
+
+        //SpellInfo const* fingerFrostInfo = sSpellMgr->GetSpellInfo(SPELL_MAGE_FINGERS_OF_FROST, GetCastDifficulty());
+
+        // Frostbolt has a 15% chance and Frozen Orb damage has a 10% to grant a charge of Fingers of Frost
+        switch (spellId) {
```
Comment (Ovahlord) [inline src/server/scripts/Spells/spell_mage.cpp:722]: ahem.

File: src/server/scripts/Spells/spell_mage.cpp [hunk 5]
```diff
@@ -695,6 +698,53 @@ class spell_mage_water_elemental_freeze : public SpellScript
...
+            case SPELL_MAGE_FROSTBOLT:
+                //chance = fingerFrostInfo->GetEffect(EFFECT_0)->CalcValue();
+                chance = 15;
+
+                // Frozen Touch (Frostbolt grants you Fingers of Frost and Brain Freeze 20% more often)
+                if (caster->HasAura(SPELL_MAGE_FROZEN_TOUCH)) {
```
Comment (Ovahlord) [inline src/server/scripts/Spells/spell_mage.cpp:728]: unneeded brackets

**Comment by Shauren** [comment]
Implemented differently, using the proc system, in b94851b824f6e4d7f7a7cd01a4c99e02b1e86b97

**Pull request closed without merging**